### DISCUSSION TITLE
book2   in cornell box , the light only illuminates one area

### DISCUSSION THREAD

**Discussion by endiow** [RayTracing/raytracing.github.io#1626] [Debugging]
The light source does not emit light in all directions

![issue](https://github.com/user-attachments/assets/0d5525bf-c8b1-405c-af18-c1c927a7cacf)

![issue1](https://github.com/user-attachments/assets/dd58f280-bb14-4462-b1e0-4bb4520a2bd3)

but the simple light scene renders right

![simple_light](https://github.com/user-attachments/assets/e2f98f7c-f378-4b19-a15b-f88c4ea2e7fd)

I really don't know where I went wrong， My code is here：https://github.com/endiow/RayTracing-The-next-week/tree/main/RayTracing_1

Can somenoe help me, pls  :(

**Comment by whydoubt** [comment]
Don't know if you ever resolved this, but I do see what happened.

Your function random_double(double min, double max) creates a **static** uniform_real_distribution.  The function is called the first time with (0, 1). When the function is later called with (-1, 1) it still uses the (0, 1) range.
